### PR TITLE
[Scripts] Basketball rack

### PR DESCRIPTION
**Pull request by imgntn**
This PR adds a script that creates a basketball rack with basketballs on it.  It also adds the rack to the master reset script for toybox, and reorganizes all the basketball related scripts to be together.

https://rawgit.com/imgntn/hifi/basketball_rack/examples/toys/basketball/createRack.js

### CODE REVIEW

File: examples/toys/basketball/createRack.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+function createCollidingBalls() {
+    var position = rackStartPosition;
+
+    var i;
+    for (i = 0; i < NUMBER_OF_BALLS; i++) {
+        var ballPosition = {
+            x: position.x,
+            y: position.y + DIAMETER * 2,
+            z: position.z + (DIAMETER) - (DIAMETER * i)
+        }
```
Comment (ZappoMan) [inline examples/toys/basketball/createRack.js]: coding standard: statements end in ;

File: examples/toys/basketball/createRack.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    collidingBalls.forEach(function(ball, index) {
+        var currentPosition = Entities.getEntityProperties(ball, "position").position;
+        var originalPosition = originalBallPositions[index];
+        var distance = Vec3.subtract(originalPosition, currentPosition);
+        var length = Vec3.length(distance);
+        if (length > RESET_DISTANCE) {
+            Script.setTimeout(function() {
+                var newPosition = Entities.getEntityProperties(ball, "position").position;
+                var moving = Vec3.length(Vec3.subtract(currentPosition, newPosition));
+                if (moving < 0.05) {
```
Comment (ZappoMan) [inline examples/toys/basketball/createRack.js]: coding standard: magic number

File: unpublishedScripts/masterReset.js [hunk 3]
```diff
@@ -521,6 +523,144 @@ function createBasketballHoop() {
...
+
+    function createCollidingBalls() {
+        var position = rackStartPosition;
+        var i;
+        for (i = 0; i < NUMBER_OF_BALLS; i++) {
+            var ballPosition = {
+                x: position.x,
+                y: position.y + DIAMETER * 2,
+                z: position.z + (DIAMETER) - (DIAMETER * i)
+            }
```
Comment (ZappoMan) [inline unpublishedScripts/masterReset.js]: coding standard: statements end in ;

File: unpublishedScripts/masterReset.js [hunk 4]
```diff
@@ -521,6 +523,144 @@ function createBasketballHoop() {
...
+                    var newPosition = Entities.getEntityProperties(ball, "position").position;
+                    var moving = Vec3.length(Vec3.subtract(currentPosition, newPosition));
+                    if (moving < 0.05) {
+                        resetCount++;
+                        if (resetCount === NUMBER_OF_BALLS) {
+                            deleteCollidingBalls();
+                            createCollidingBalls();
+                        }
+                    }
+                }, 200)
```
Comment (ZappoMan) [inline unpublishedScripts/masterReset.js]: coding standard: statements end in ;

**Comment by imgntn** [comment]
thanks @ZappoMan

**Pull request merged**